[CI] InstrumentsConcurrencyTests testLockingWhenRegistering failing

Looks like https://github.com/elastic/elasticsearch/issues/100251 but on 8.11, maybe the fix just needs backporting.

**Build scan:**
https://gradle-enterprise.elastic.co/s/54xhyw4qoetdi/tests/:modules:apm:test/org.elasticsearch.telemetry.apm.internal.metrics.InstrumentsConcurrencyTests/testLockingWhenRegistering
**Reproduction line:**
```
./gradlew ':modules:apm:test' --tests "org.elasticsearch.telemetry.apm.internal.metrics.InstrumentsConcurrencyTests.testLockingWhenRegistering" -Dtests.seed=58592E77DE056E97 -Dtests.locale=sr -Dtests.timezone=Pacific/Port_Moresby -Druntime.java=17
```

**Applicable branches:**
8.11

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.telemetry.apm.internal.metrics.InstrumentsConcurrencyTests&tests.test=testLockingWhenRegistering
**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <WAITING>
     but: was <TERMINATED>

  at __randomizedtesting.SeedInfo.seed([58592E77DE056E97:667D2805D3682E21]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.telemetry.apm.internal.metrics.InstrumentsConcurrencyTests.lambda$testLockingWhenRegistering$2(InstrumentsConcurrencyTests.java:102)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1196)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1169)
  at org.elasticsearch.telemetry.apm.internal.metrics.InstrumentsConcurrencyTests.testLockingWhenRegistering(InstrumentsConcurrencyTests.java:102)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/modules/apm/src/test/java/org/elasticsearch/telemetry/apm/internal/metrics/InstrumentsConcurrencyTests.java
+++ b/modules/apm/src/test/java/org/elasticsearch/telemetry/apm/internal/metrics/InstrumentsConcurrencyTests.java
@@ -90,7 +90,6 @@ public class InstrumentsConcurrencyTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/100587")
     public void testLockingWhenRegistering() throws Exception {
         Instruments instruments = new Instruments(lockingMeter);
 

--- a/modules/apm/src/test/java/org/elasticsearch/telemetry/apm/internal/metrics/InstrumentsConcurrencyTests.java
+++ b/modules/apm/src/test/java/org/elasticsearch/telemetry/apm/internal/metrics/InstrumentsConcurrencyTests.java
@@ -28,12 +28,13 @@ import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.sameInstance;
 
 public class InstrumentsConcurrencyTests extends ESTestCase {
-    String name = "name";
-    String description = "desc";
-    String unit = "kg";
-    Meter noopMeter = OpenTelemetry.noop().getMeter("noop");
-    CountDownLatch registerLatch = new CountDownLatch(1);
-    Meter lockingMeter = new Meter() {
+    private final String name = "name";
+    private final String description = "desc";
+    private final String unit = "kg";
+    private final Meter noopMeter = OpenTelemetry.noop().getMeter("noop");
+    private final CountDownLatch buildLatch = new CountDownLatch(1);
+    private final CountDownLatch registerLatch = new CountDownLatch(1);
+    private final Meter lockingMeter = new Meter() {
         @Override
         public LongCounterBuilder counterBuilder(String name) {
             return new LockingLongCounterBuilder();
@@ -75,6 +76,7 @@ public class InstrumentsConcurrencyTests extends ESTestCase {
         @Override
         public LongCounter build() {
             try {
+                buildLatch.countDown();
                 registerLatch.await();
             } catch (Exception e) {
                 throw new RuntimeException(e);
@@ -95,6 +97,8 @@ public class InstrumentsConcurrencyTests extends ESTestCase {
         var registerThread = new Thread(() -> instruments.registerLongCounter(name, description, unit));
         // registerThread has a countDown latch that is simulating a long-running registration
         registerThread.start();
+        buildLatch.await(); // wait for registerThread to hold the lock
+
         var setProviderThread = new Thread(() -> instruments.setProvider(noopMeter));
         // a setProviderThread will attempt to override a meter, but will wait to acquireLock
         setProviderThread.start();

